driver-core/src/test/resources/connection-string/valid-warnings.json

```diff
@@ -63,6 +63,51 @@
       "options": {
         "wtimeoutms": 10
       }
+    },
+    {
+      "description": "Empty integer option values are ignored",
+      "uri": "mongodb://localhost/?maxIdleTimeMS=",
+      "valid": true,
+      "warning": true,
+      "hosts": [
+        {
+          "type": "hostname",
+          "host": "localhost",
+          "port": null
+        }
+      ],
+      "auth": null,
+      "options": null
+    },
+    {
+      "description": "Empty boolean option value are ignored",
+      "uri": "mongodb://localhost/?journal=",
+      "valid": true,
+      "warning": true,
+      "hosts": [
+        {
+          "type": "hostname",
+          "host": "localhost",
+          "port": null
+        }
+      ],
+      "auth": null,
+      "options": null
+    },
+    {
+      "description": "Comma in a key value pair causes a warning",
+      "uri": "mongodb://localhost?authMechanism=MONGODB-OIDC&authMechanismProperties=TOKEN_RESOURCE:mongodb://host1%2Chost2",
```

Comment:
This test now fails so it's skipped in code.  Previously it was only passing due to the missing `authMechanism` property, which was causing the `authMechanismProperties` to be ignored.  Now that they are not ignored, this connection string throws an exception at this spot:

```
                if (mechanismPropertyKeyValue.length != 2) {
                    throw new IllegalArgumentException(format("The connection string contains invalid authentication properties. "
                            + "'%s' is not a key value pair", part));
                }
```

The message is`'host2' is not a key value pair`.

Should that be filed as a bug?

Reply:
I think so.

It seems we should notify the user via an error if there is a possible user error, such as forgetting a value. However, the spec says that we ignore at least missing numbers (I don't know the rationale for this - perhaps forward-compatibility?). The spec (according to this test) now also says that it is ok to have a stray mechanism property (called `host2`). I'm not sure this should be allowed, at the spec level - what if the user is trying to enable a security feature that we do not support in an earlier version? It seems that we should fail. In any case, there's either a bug in the spec, or a (minor) bug in the driver.

Reply:
Will file

Reply:
Filed JAVA-5504